File: 1963-minimum-number-of-swaps-to-make-the-string-balanced/1963-minimum-number-of-swaps-to-make-the-string-balanced.java
class Solution {
    public int minSwaps(String s) {
        Stack<Character> stack = new Stack();
        char[] chars = s.toCharArray();
        for (char c : chars) {
            if (!stack.isEmpty() && c == ']') {
                if (stack.peek() == '[') stack.pop();
            } else {
                stack.add(c);
            }
        }
        return (int)Math.ceil(stack.size() / 2);
    }
}

File: 1963-minimum-number-of-swaps-to-make-the-string-balanced/1963-minimum-number-of-swaps-to-make-the-string-balanced.kt
class Solution {
fun minSwaps(s: String): Int {
    val st = Stack<Char>()
    var count = 0
    val n = s.length
    for (i in 0 until n) {
        if (s[i] == ']') {
            if (!st.isEmpty() && st.peek() == '[') {
                st.pop()
            } else {
                count++
            }
        } else {
            st.push(s[i])
        }
    }
    return (count + 1) / 2
}
}